pcs/pcs.go

```diff
@@ -158,14 +160,27 @@ type PckCertTCB struct {
 }
 
 // PckExtensions represents the information stored in the x509 extensions of a PCK certificate which
-// will be required for verification
+// will be required for verification/validation
 type PckExtensions struct {
-	PPID  string
-	TCB   PckCertTCB
-	PCEID string
-	FMSPC string
+	PPID    string
+	TCB     PckCertTCB
+	PCEID   string
+	FMSPC   string
+	SGXType SGXType
 }
 
+// SGXType represents the type of the platform for which the PCK certificate was created
+type SGXType int
```

Comment:
I'm not quite familiar with the "SGXType" concept. Is it something Intel defined? Can you provide a link to it?

Reply:
Yes, this is a concept by Intel. The values are defined in [Intel® SGX PCK Certificate and Certificate Revocation List Profile Specification 1.5](https://api.trustedservices.intel.com/documents/Intel_SGX_PCK_Certificate_CRL_Spec-1.5.pdf), 1.3.5 Intel® SGX PCK Certificate, SGX Type:

![image](https://github.com/user-attachments/assets/6812b6d6-6871-49da-a66a-8e8f51d5680c)

The difference between `SGXTypeScalable` and `SGXTypeScalableWithIntegrity` is explained in the Intel® Trust Domain Extensions white paper, 02. Intel TDX – Technical Explanation:
> The SEAM-memory range [...] can operate in one of two available modes for memory integrity protection (to enable various memory configurations). Memory integrity may be enforced by either (the default) cryptographic-integrity protection scheme or a logical-integrity protection scheme. The cryptographic-integrity scheme uses a SHA-3-based, message-authentication code (MAC) (28-bit) that helps prevent host/system software accesses as well as detects state-tamper from software (e.g., rowhammer) and some hardware attacks. The logical-integrity protection scheme is designed to prevent host/system software accesses only.

and Intel® Trust Domain Extensions white paper, 02. Intel TDX – Technical Explanation, E. REMOTE ATTESTATION:
> Platform configuration differences such as whether cryptographic vs. logical-integrity mode is enabled are reflected via the Provisioning Certificates.

I wasn't able to find any docs that directly state that the SGX type is the field recording this platform configuration difference, but I was able to confirm this experimentally by generating two quotes, one on a platform with logical integrity and one on a platform with cryptographic integrity, and comparing the values.